src/coreclr/jit/ssabuilder.cpp

```diff
@@ -860,95 +860,65 @@ void SsaBuilder::AddDefToEHSuccessorPhis(BasicBlock* block, unsigned lclNum, uns
-                        assert(phiDef->IsPhiDefn());
+                // skip the phi node to get to the GT_STORE_LCL_VAR
+                if (tree->OperIs(GT_PHI_ARG, GT_PHI))
+                {
+                    continue;
+                }
 
-                        if (phiDef->GetLclNum() == lclNum)
-                        {
-                            // It's the definition for the right local.  Add "ssaNum" to the RHS.
-                            AddPhiArg(handler, stmt, phiDef->Data()->AsPhi(), lclNum, ssaNum, block);
+                assert(tree->OperIs(GT_STORE_LCL_VAR));
+                if (tree->AsLclVarCommon()->GetLclNum() == lclNum)
+                {
+                    AddPhiArg(succ, nullptr, tree->gtGetOp1()->AsPhi(), lclNum, ssaNum, block);
 #ifdef DEBUG
-                            phiFound = true;
+                    phiFound = true;
 #endif
-                            break;
-                        }
-                    }
-#if defined(TARGET_WASM)
                 }
-#endif
-                assert(phiFound);
             }
-
-            unsigned nextTryIndex = tryBlk->ebdEnclosingTryIndex;
-            if (nextTryIndex == EHblkDsc::NO_ENCLOSING_INDEX)
-=======
-        // A prefix of blocks statements will be SSA definitions.  Search those for "lclNum".
-        for (Statement* const stmt : succ->Statements())
+        }
+        else
         {
-            // If the tree is not an SSA def, break out of the loop: we're done.
-            if (!stmt->IsPhiDefnStmt())
->>>>>>> origin/runtime-main
+#endif
+            // A prefix of blocks statements will be SSA definitions.  Search those for "lclNum".
+            for (Statement* const stmt : succ->Statements())
             {
-                break;
-            }
+                // If the tree is not an SSA def, break out of the loop: we're done.
+                if (!stmt->IsPhiDefnStmt())
+                {
+                    break;
+                }
 
-            GenTreeLclVar* phiDef = stmt->GetRootNode()->AsLclVar();
-            assert(phiDef->IsPhiDefn());
+                GenTreeLclVar* phiDef = stmt->GetRootNode()->AsLclVar();
+                assert(phiDef->IsPhiDefn());
 
-            if (phiDef->GetLclNum() == lclNum)
-            {
-                // It's the definition for the right local.  Add "ssaNum" to the RHS.
-                AddPhiArg(succ, stmt, phiDef->Data()->AsPhi(), lclNum, ssaNum, block);
+                if (phiDef->GetLclNum() == lclNum)
+                {
+                    // It's the definition for the right local.  Add "ssaNum" to the RHS.
+                    AddPhiArg(succ, stmt, phiDef->Data()->AsPhi(), lclNum, ssaNum, block);
 #ifdef DEBUG
-                phiFound = true;
+                    phiFound = true;
 #endif
-                break;
+                    break;
+                }
             }
+#if defined(TARGET_WASM)
         }
+#endif
```

Comment:
Nit: could you rebalance the `#ifdef` like so:
```
#ifdef TARGET_WASM
  if (block->IsLIR()
  {
  }
  else
#endif // TARGET_WASM
  {
  }
```
It is easier to read than:
```
#ifdef TARGET_WASM
  if (block->IsLIR()
  {
  }
  else
  {
#endif // TARGET_WASM

#ifndef TARGET_WASM
  }
#endif
```

Reply:
thanks

Reply:
Okay thank you